controllers/application/certificate.go

```diff
@@ -62,33 +56,34 @@ func (r *ApplicationReconciler) reconcileCertificate(ctx context.Context, applic
 		}
 	}
 
-	// Clear out unused certificates
+	// Clear out unused certs
 	certificates := certmanagerv1.CertificateList{}
 	err := r.GetClient().List(ctx, &certificates, client.InNamespace("istio-system"))
 	if err != nil {
 		r.SetControllerError(ctx, application, controllerName, err)
 		return reconcile.Result{}, err
 	}
 
-	for i := range certificates.Items {
-		certificate := &certificates.Items[i]
+	for _, certificate := range certificates.Items {
 
-		// Skip unrelated certificates matching string namespace-name-*
-		match, _ := regexp.MatchString("^"+application.Namespace+"-"+application.Name+"-.*$", certificate.Name)
-		// println(certificate.Name + ", matches pattern: " + strconv.FormatBool(match))
-
-		if match {
+		certificateInApplicationSpecIndex := slices.IndexFunc(application.Spec.Ingresses, func(hostname string) bool {
+			certificateName := fmt.Sprintf("%s-%s-ingress-%x", application.Namespace, application.Name, util.GenerateHashFromName(hostname))
+			return certificate.Name == certificateName
+		})
+		certificateInApplicationSpec := certificateInApplicationSpecIndex != -1
+		if certificateInApplicationSpec {
 			continue
 		}
 
-		// Skip active certificates
-		_, ok := active[certificate.Name]
-		if ok {
+		// We want to delete certificate which are not in the spec, but still "owned" by the application.
+		applicationNameInCertificate := strings.Contains(certificate.Name, application.Name)
+		applicationNamespaceInCertificate := strings.Contains(certificate.Name, application.Namespace)
```

Comment:
Her ser det ut som du kan gå på en smell om f.eks. namespacet heter dsa og applikasjonen heter dsa-frontend. Siden den kun sjekker om strengen finnes i navnet uavhengig av hva det er vil du potensielt få uventet oppførsel hvor en applikasjon vil matche på andre applikasjoners certificates siden de inneholder navnet på namespacet i application.name slik at `applicationNameInCertificate` blir true. 

Bedre å bruke regex her?

Reply:
jeg tror ikke det ville forårsaket en feil i namespace: dsa og name: dsa-frontend tilfellet, men ja det vil nok være lurt å matche på hele namespace-name over individuelt, ettersom begge kan inneholde hverandre i forskjellige tilfeller